.github/workflows/release-plugin.yml

```diff
@@ -1,33 +1,233 @@
+    needs: validate_version
+    if: ${{ github.event.inputs.skip_tests == 'false' }}
+    uses: ./.github/workflows/integration-tests.yml
+
+  e2e_tests:
+    name: End-to-end Tests
+    needs: validate_version
+    if: ${{ github.event.inputs.skip_tests == 'false' }}
+    uses: ./.github/workflows/e2e-tests.yml
+
+  tag_and_release:
+    name: Tag and Release
+    needs: [ validate_version, integration_tests, e2e_tests ]
+    if: ${{ !failure() && !cancelled() }}
+    runs-on: ubuntu-latest
+    outputs:
+      tag_name: ${{ steps.tag.outputs.tag_name }}
+    steps:
+      - name: Configure Git
+        run: |
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+
+      - name: Checkout the specific branch/ref
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.SOURCE_REF }}
+          fetch-depth: 0
+
+      - name: Tag the release
+        id: tag
+        run: |
+          TAG_NAME="${{ env.VERSION }}"
+          if ${{ env.DRY_RUN }}; then
+            TAG_NAME="dry-run-${{ env.VERSION }}"
+            # If the tag already exists, delete the tag
+            if git tag --list | grep -q "${TAG_NAME}"; then
+              git tag -d "${TAG_NAME}"
+            fi
+          fi
+
+          # Set the tag name as an output
+          echo "TAG_NAME=${TAG_NAME}" >> $GITHUB_ENV
+          echo "TAG_NAME=${TAG_NAME}" >> $GITHUB_OUTPUT
+
+          git tag "${TAG_NAME}"
+          git push origin "${TAG_NAME}"
+
+          echo "Tagged release with tag '${TAG_NAME}'"
+
+      - name: Print changelog
+        run: |
+          cat CHANGELOG.md
+
+      - name: Extract Changelog
+        id: extract_changelog
+        run: |
+          set -e
+          VERSION=${{ env.VERSION }}
+          START_LINE=$(grep -n "## \[${VERSION}\]" CHANGELOG.md | cut -d: -f1)
+          if [ -z "$START_LINE" ]; then
+            echo "Version not found in CHANGELOG.md" >&2
+            exit 1
+          fi
+          TAIL_LINE=$(tail -n +$((START_LINE + 1)) CHANGELOG.md | grep -n "^## " | head -n 1 | cut -d: -f1 || true)
+          if [ -z "$TAIL_LINE" ]; then
+            END_LINE=$(wc -l < CHANGELOG.md)
+          else
+            END_LINE=$((START_LINE + TAIL_LINE - 1))
+          fi
+          sed -n "${START_LINE},${END_LINE}p" CHANGELOG.md | sed '$d' > release_notes.md
+          cat release_notes.md
+        shell: bash
+
+      - name: Format Changelog
+        id: format_changelog
+        run: |
+          sed -i '1d' release_notes.md # Remove the first line
+          sed -i 's/###/##/g' release_notes.md # Change headers from ### to ##
+        shell: bash
+
+      - name: Create a GitHub release
+        id: github_release
+        uses: ncipollo/release-action@v1
+        with:
+          tag: ${{ steps.tag.outputs.tag_name }}
+          name: ${{ env.VERSION }}
+          bodyFile: ./release_notes.md
+          draft: true
```

Comment:
Why do we publish the GitHub release as a draft? Do I correctly see that this gets updated afterwards?

Reply:
Ah yes, it initially creates the release as a draft, and in the "Update release with ZIP file" step it updates it to a regular release (if dry-run is false). This is so the release is not publicly visible until the full workflow finishes running.